template/pdk.gen.go.ejs

```diff
@@ -105,7 +105,8 @@ func _<%- goName(ex.name) %>() int32 {
 		<% if (p.description) { -%>
 		// <%- formatCommentBlock(p.description, "// ") %>
 		<% } -%>
-		<%- goName(p.name) %> <%- p.nullable ? pointerToGolangType(p) : toGolangType(p) %> `json:"<%- p.name %><%- p.nullable ? ",omitempty" : null %>"`
+    <% let isPointer = p.nullable || !p.required %>
+		<%- goName(p.name) %> <%- isPointer ? pointerToGolangType(p) : toGolangType(p) %> `json:"<%- p.name %><%- !p.required ? ",omitempty" : null %>"`
```

Comment:
We need a pointer if it's either optional or it's nullable

We use `omitempty` only when it's optional